firmware/targets/f7/Src/update.c

```diff
@@ -0,0 +1,166 @@
+#include "update.h"
+
+#include <furi.h>
+#include <furi_hal.h>
+#include <flipper.h>
+
+#include <fatfs.h>
+#include <flipper_format/flipper_format.h>
+
+#include <updater/util/update_manifest.h>
+
+static FATFS fs;
+
+static const char FS_ROOT_PATH[] = "/";
+static const char CONFIG_PATH[] = UPDATE_MAINFEST_DEFAULT_PATH;
+
+#define CHECK_FRESULT(result)   \
+    {                           \
+        if((result) != FR_OK) { \
+            return false;       \
+        }                       \
+    }
+
+static bool flipper_update_init() {
+    furi_hal_interrupt_init();
+    furi_hal_delay_init(false);
+
+    MX_GPIO_Init();
+
+    //furi_hal_bootloader_init();
+    //furi_hal_version_init();
+
+    furi_hal_spi_init();
+    furi_hal_crc_init();
+
+    MX_FATFS_Init();
+    if(!hal_sd_detect()) {
+        return false;
+    }
+
+    if(BSP_SD_Init(true)) {
+        return false;
+    }
+
+    CHECK_FRESULT(f_mount(&fs, FS_ROOT_PATH, 1));
+    return true;
+}
+
+static inline void flipper_update_target_switch(void* offset) {
+    asm volatile("ldr    r3, [%0]    \n"
+                 "msr    msp, r3     \n"
+                 "ldr    r3, [%1]    \n"
+                 "mov    pc, r3      \n"
+                 :
+                 : "r"(offset), "r"(offset + 0x4)
```

Comment:
Need to check VTOR OFFSET calculation

Reply:
there are no extra offsets - VTOR starts at ram base, params are fed through VECT_TAB_OFFSET & other definitions